internal/collections/set.go

```diff
@@ -0,0 +1,33 @@
+package collections
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Set is a container that can hold each item only once and has a fast lookup time.
+//
+// You can define a new set like this:
+//
+//	var validKeyLengths = golang.Set[int]{
+//	    16: {},
+//	    24: {},
+//	    32: {},
+//	}
```

Comment:
I would be tempted too, but performance-wise using the `NewSet` is less than optimal since it needs to iterate over all items, whereas directly setting the values doesn't have any performance cost.

Reply:
Do we really have to worry about this level of performance of a small Set package here?

Reply:
@Yantrio would you mind adding this into the branch as a commit?

Reply:
I can take this